### DISCUSSION TITLE
Convergence/self heating issue

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#590] [Q&A]
Hello, 

I'm running a 1D background plasma simulation and scanning different numbers of macroparticles using the deck attached below. However, I'm getting an issue with self-heating or convergence issues. My result is quiet dependent on the number of macroparticles used (please see the image attached) even for a very large number of macroparticles (npart = 200 *nx, which is higher than anything I have seen in the input deck examples). Can I please get some insight on why this is the case or how to mitigate this? I'm also running with higher particle shape order and current smoothing. 

(For reference, the cell size dx is roughly the same order of magnitude as the Debye length) 

[input.txt](https://github.com/Warwick-Plasma/epoch/files/13402353/input.txt)

![image](https://github.com/Warwick-Plasma/epoch/assets/108578700/a4a4795a-5667-4a41-85a2-b038f336a230)

**Comment by Status-Mirror** [comment]
Hey William,

The example decks are to demonstrate coding syntax, and are designed to be run on small computers. Often, PIC simulations can require very high resolution to produce results which do not suffer from self-heating, and you may want to look into running on cluster PCs for main results.

We have a guide on ways to mitigate self-heating on our [basic examples](https://epochpic.github.io/quickstart/basic_examples/particle_load.html#self-heating) page. You may find the content covered here useful.

Cheers,
Stuart

**Reply by cooluser26** [reply]
Hi Stuart,

Thanks for your reply again. The main issue is that it starts taking too long to run, even for the background plasma-only case. The idea is to eventually use the parameters here to propagate a beam through a plasma. But I need to ensure the self-heating isn't too much of an issue. 

I can make the simulation here smaller for testing purposes, but I need a much larger domain once the beam is included. Would that issue be solved with a moving window? So, the window would move with the beam, and I wouldn't need to make the simulation domain larger.

Best,

William.

**Reply by Status-Mirror** [reply]
A [moving window](https://epochpic.github.io/documentation/input_deck/input_deck_window.html) could be used. There seems to be a lot of interest in moving windows currently, so I might try and create a more detailed demo for this in our basic examples.

When you include a moving window, cells are added onto the end of the simulation window. These are filled with macro-particles based on the initial conditions of your code, so they would enter the simulation window at the correct temperature and it would take longer for them to undergo self-heating. Using fewer cells would also allow you to model a higher ppc, further reducing self-heating.

Cheers,
Stuart

**Reply by cooluser26** [reply]
Yes please! A demo in moving windows would be extremely helpful! (I will probably just try to figure out how it all works with the current examples in the documentation first)

**Comment by amarsarari** [comment]
Hi wnguyen1312  could you please send me matlab or python script for energy vs time   and    energy density vs time.
